From 4503ab12984bc093059c2436070146cf9b8f1dcd Mon Sep 17 00:00:00 2001
From: aken_liu <aken_liu@accton.com.tw>
Date: Fri, 14 Dec 2018 18:12:34 +0800
Subject: [PATCH] support LM75 on the PIU we need to create the lm75's sysfs
 whatever the PIU is inserted or not.
 Manually MODIFIED for kern 4.1 by <Peter.Bugni@us.fujitsu.com>

---
 drivers/hwmon/Kconfig | 8 ++++++++
 drivers/hwmon/lm75.c  | 5 ++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/hwmon/Kconfig b/drivers/hwmon/Kconfig
index a6905ed..98a42f1 100644
--- a/drivers/hwmon/Kconfig
+++ b/drivers/hwmon/Kconfig
@@ -691,6 +691,14 @@ config SENSORS_LM75
 	  This driver can also be built as a module.  If so, the module
 	  will be called lm75.
 
+config SENSORS_LM75_ON_PIU
+        tristate "National Semiconductor LM75 on the PIU"
+        depends on I2C
+        help
+          If you say yes here you get support for National Semiconductor LM75
+          sensor chips on the PIU.
+
+
 config SENSORS_LM77
 	tristate "National Semiconductor LM77"
 	depends on I2C
diff --git a/drivers/hwmon/lm75.c b/drivers/hwmon/lm75.c
index c03b490..b5fa9bb 100644
--- a/drivers/hwmon/lm75.c
+++ b/drivers/hwmon/lm75.c
@@ -286,7 +286,10 @@ lm75_probe(struct i2c_client *client, const struct i2c_device_id *id)
 	status = lm75_read_value(client, LM75_REG_CONF);
 	if (status < 0) {
 		dev_dbg(dev, "Can't read config? %d\n", status);
-		return status;
+        
+#ifndef CONFIG_SENSORS_LM75_ON_PIU /* We sould not retuen error for hot-plug lm75 from PIU. */
+		return status; 
+#endif
 	}
 	data->orig_conf = status;
 	new = status & ~clr_mask;
-- 
1.9.1

